app/controllers/mixins/custom_buttons.rb

```diff
@@ -24,6 +24,8 @@ def custom_toolbar_simple
       Mixins::CustomButtons::Result.new(:single)
     elsif @lastaction == "show_list"
       Mixins::CustomButtons::Result.new(:list)
+    elsif @display == 'generic_objects'
```

Comment:
So this is something that is going to be generic? This is a generic method used in many controllers.

If the generic objects are to be used on many places than ok. Otherwise the changes should be done in a redefined method in a controller class.